### PR TITLE
fix: removed logger from NonCPS to CPS

### PR DESCRIPTION
**Pull request by petod108**
moved logger from NonCPS to upper CPS function.
If logger is used in NonCPS then there is error:

DEBUG: Parsing jira data to document history
expected to call org.ods.orchestration.util.DocumentHistory.computeEntryData but wound up catching org.ods.util.Logger.warn; see: https://jenkins.io/redirect/pipeline-cps-method-mismatches/
...
WARN: Error occured within the orchestration pipeline: Error: Creating document of type 'CFTR' for project 'tes96' in phase 'Test' and stage 'PRE_END' has failed: Could not find matching constructor for: org.ods.orchestration.util.DocumentHistoryEntry(java.lang.String, java.lang.Long, java.lang.String, java.lang.String, java.lang.String)

### CODE REVIEW

**Comment by metmajer** [comment]
@s2oBCN @petod108 an invalid document history…. Not good and a warning might not be enough. What’s the reason for this case? How can a user fix it?

**Review by clemensutschig — DISMISSED** [review]

**Comment by petod108** [comment]
> @s2oBCN @petod108 an invalid document history…. Not good and a warning might not be enough. What’s the reason for this case? How can a user fix it?
En JiraUseCase for new reporting bug,  there was not added versions for new reported Bug to the JiraDataItem.
This then leaded to this Warning logging. Because in sequence:
* load JiraDataItem
* report bug, but do not fill versions in JiraDataItem
* check if all items have versions (fail)

**Review by clemensutschig — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ## Unreleased
 
 ## [4.0] - 2021-05-11
-
+- Prevent Jenkins nonCPS error after reporting bug  ([#776](https://github.com/opendevstack/ods-jenkins-shared-library/pull/776))
```
Comment (s2oBCN) [inline CHANGELOG.md:6]: Please move this line above , is not for 4.0

File: src/org/ods/orchestration/util/DocumentHistory.groovy [hunk 2]
```diff
@@ -325,6 +326,9 @@ class DocumentHistory {
         this.allIssuesAreValid = true
 
         def versionMap = this.computeEntryData(jiraData, projectVersion, keysInDocument)
+        if (!this.allIssuesAreValid){
```
Comment (s2oBCN) [inline src/org/ods/orchestration/util/DocumentHistory.groovy:329]: let me talk online about this code

File: CHANGELOG.md (hunk 1 above)
Comment (s2oBCN) [inline CHANGELOG.md:6]: Please move this above, this is not 4.x

**Review by s2oBCN — APPROVED** [review]

**Pull request merged**